bindings/swift/Sources/FoundationDB/FoundationdDB.swift

```diff
@@ -0,0 +1,187 @@
+    /// Gets the read version used by this transaction
+    func getReadVersion() async throws -> Int64
+}
+
+public extension ITransaction {
+    func getValue(for key: String, snapshot: Bool = false) async throws -> Fdb.Value? {
+        let keyBytes = [UInt8](key.utf8)
+        return try await getValue(for: keyBytes, snapshot: snapshot)
+    }
+
+    func getValue(for key: Fdb.Key, snapshot: Bool = false) async throws -> Fdb.Value? {
+        try await getValue(for: key, snapshot: snapshot)
+    }
+
+    func setValue(_ value: String, for key: String) {
+        let keyBytes = [UInt8](key.utf8)
+        let valueBytes = [UInt8](value.utf8)
+        setValue(valueBytes, for: keyBytes)
+    }
+
+    func clear(key: String) {
+        let keyBytes = [UInt8](key.utf8)
+        clear(key: keyBytes)
+    }
+
+    func clearRange(beginKey: String, endKey: String) {
+        let beginKeyBytes = [UInt8](beginKey.utf8)
+        let endKeyBytes = [UInt8](endKey.utf8)
+        clearRange(beginKey: beginKeyBytes, endKey: endKeyBytes)
+    }
+
+    func getKey(selector: Fdb.Selectable, snapshot: Bool = false) async throws -> Fdb.Key? {
+        try await getKey(selector: selector.toKeySelector(), snapshot: snapshot)
+    }
+
+    func getKey(selector: Fdb.KeySelector, snapshot: Bool = false) async throws -> Fdb.Key? {
+        try await getKey(selector: selector, snapshot: snapshot)
+    }
+
+    func getRange(
+        begin: Fdb.Selectable, end: Fdb.Selectable, limit: Int32 = 0, snapshot: Bool = false
+    ) async throws -> ResultRange {
+        let beginSelector = begin.toKeySelector()
+        let endSelector = end.toKeySelector()
+        return try await getRange(
+            beginSelector: beginSelector, endSelector: endSelector, limit: limit, snapshot: snapshot
+        )
+    }
+
+    func getRange(
+        beginSelector: Fdb.KeySelector, endSelector: Fdb.KeySelector, limit: Int32 = 0,
+        snapshot: Bool = false
+    ) async throws -> ResultRange {
+        try await getRange(
+            beginSelector: beginSelector, endSelector: endSelector, limit: limit, snapshot: snapshot
+        )
+    }
+
+    func getRange(
+        beginKey: String, endKey: String, limit: Int32 = 0, snapshot: Bool = false
+    ) async throws -> ResultRange {
+        let beginKeyBytes = [UInt8](beginKey.utf8)
+        let endKeyBytes = [UInt8](endKey.utf8)
+        return try await getRange(
+            beginKey: beginKeyBytes, endKey: endKeyBytes, limit: limit, snapshot: snapshot
+        )
+    }
+
+    func getRange(
+        beginKey: Fdb.Key, endKey: Fdb.Key, limit: Int32 = 0, snapshot: Bool = false
+    ) async throws -> ResultRange {
+        try await getRange(beginKey: beginKey, endKey: endKey, limit: limit, snapshot: snapshot)
+    }
+}
+
+public extension IDatabase {
+    func withTransaction<T: Sendable>(
+        _ operation: (ITransaction) async throws -> T
+    ) async throws -> T {
+        let maxRetries = 100 // TODO: Remove this.
```

Comment:
Remove this. Do what is consistent with other bindings. Can add other method to do with specific number of retries.